contents/docs/product-analytics/sql.md

```diff
@@ -0,0 +1,293 @@
+
+### FROM
+
+Use `FROM` to select the database table to run the queries on. In PostHog, examples include `events`, `groups`, `raw_session_replay_events`, and more listed in the [data management database tab](https://app.posthog.com/data-management/database) in your PostHog instance.
+
+```sql
+SELECT session_id, min_first_timestamp, click_count 
+FROM raw_session_replay_events
+```
+
+You can join multiple tables together using the `LEFT JOIN` command which takes one table before the command and another after the command and combines them based on the join condition using the `ON` keyword.
+
+```sql
+SELECT events.event, persons.is_identified
+FROM events
+LEFT JOIN persons ON events.person_id = persons.id
+```
+
+### WHERE
+
+Use `WHERE` to filter rows based on specified conditions. These conditions can be:
+
+1. Comparison operators like `=`, `<`, or `≤`
+2. Logical operators like `AND`, `OR`, or `NOT`. These are often used to combine multiple conditions. 
+3. Functions like `toDate`, `today()`
+4. Clauses like `LIKE`, `IN`, `IS NULL`, `BETWEEN`
+
+```sql
+SELECT *
+FROM events
+WHERE event = '$pageview'
+   AND toDate(timestamp) = today()
+   AND properties.$current_url LIKE '%/blog%'
+```
+
+### GROUP BY
+
+Use `GROUP BY` to group rows that have the same values in specified columns into summary rows.
+
+```sql
+select 
+   properties.$os,
+   count()
+from events
+group by 
+   properties.$os
+```
+
+### ORDER BY
+
+Use `ORDER BY` to sort the query results by one or more columns. You can specify order by ascending with `ASC` or descending with `DESC`.
+
+```sql
+SELECT 
+   properties.$browser, 
+   count()
+FROM events
+GROUP BY properties.$browser
+ORDER BY count() DESC
+```
+
+### AS
+
+Use `AS` to alias columns or tables with different names. This makes the query and results more readable.
+
+```sql
+SELECT 
+   properties.$current_url as current_url,
+   count() as current_url_count
+FROM events
+GROUP BY current_url
+ORDER BY current_url_count DESC
+```
+
+### LIMIT
+
+Use `LIMIT` to restrict the number of rows returned by the query. It specifies the maximum number of rows the query should retrieve. By default, PostHog sets it at `100`.
+
+```sql
+SELECT 
+   properties.$lib as library,
+   count() as library_count
+FROM events
+WHERE properties.$lib != ''
+GROUP BY library
+ORDER BY library_count DESC
+LIMIT 1
+```
+
+### HAVING
+
+Use `HAVING` with the `GROUP BY` command to filter the results based on aggregate function values. While `WHERE` filters rows before grouping, `HAVING` filters grouped results after aggregation.
+
+```sql
+SELECT 
+   properties.$os,
+   count()
+FROM events
+GROUP BY 
+   properties.$os
+HAVING count() > 100
+```
+
+### WITH
+
+Use `WITH` to define a temporary result set that you can reference within a larger query. It helps break down complex queries into smaller parts. You can think of it as a temporary table and is similar to using a subquery in a `FROM` command.
```

Comment:
Not sure if worth adding the technical fact, that the subqueries in `with` are just inlined when composing the final query. The nuance is that if you use a complex with subquery 3 times, we'll query the data 3 times. So in that sense, these are not like a temporary table... but more like a function that returns during query building time. 🤷